neo/SmartContract/ScriptFile.cs

```diff
@@ -0,0 +1,89 @@
+﻿using Neo.IO;
+using System;
+using System.IO;
+
+namespace Neo.SmartContract
+{
+    public class ScriptFile : ISerializable
+    {
+        public enum ScriptEngine : int
+        {
+            NeoVM = 0x4E454F00 // NEO\0
+        }
+
+        /// <summary>
+        /// Engine
+        /// </summary>
+        public ScriptEngine Engine { get; set; }
+
+        /// <summary>
+        /// Compiler
+        /// </summary>
+        public string Compiler { get; set; }
+
+        /// <summary>
+        /// Version
+        /// </summary>
+        public Version Version { get; set; }
+
+        /// <summary>
+        /// Script
+        /// </summary>
+        public byte[] Script { get; set; }
+
+        /// <summary>
+        /// Script Hash
+        /// </summary>
+        public UInt160 ScriptHash { get; set; }
+
+        public int Size =>
+            sizeof(ScriptEngine) +      // Engine
+            Compiler.GetVarSize() +     // Compiler
+            (sizeof(int) * 4) +         // Version
+            Script.GetVarSize() +       // Script
+            ScriptHash.Size;            // ScriptHash (CRC)
+
+        /// <summary>
+        /// Read Script Header from a binary
+        /// </summary>
+        /// <param name="script">Script</param>
+        /// <returns>Return script header</returns>
+        public static ScriptFile FromByteArray(byte[] script)
+        {
+            using (var stream = new MemoryStream(script))
+            using (var reader = new BinaryReader(stream))
+            {
+                return reader.ReadSerializable<ScriptFile>();
+            }
+        }
+
+        public void Serialize(BinaryWriter writer)
+        {
+            writer.Write((int)Engine);
+            writer.WriteVarString(Compiler);
```

Comment:
Fixed string is better. This way the header can have a fixed length.

Reply:
64?

Reply:
32 is enough.